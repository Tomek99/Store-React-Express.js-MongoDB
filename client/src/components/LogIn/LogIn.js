import React, { useContext } from "react";
import { useState } from "react";
import styles from "./LogIn.module.scss";
import { Link } from "react-router-dom";
import Benefits from "../Benefits/Benefits";
import RegisterReference from "./RegisterReference/RegisterReference";
import { Context } from "../../Contexts/Context";
import { Formik, Form, Field } from "formik";
import ErrorMessage from "../ErrorMessage/ErrMessage";
import * as Yup from "yup";
import axios from "axios";

const validationSchema = Yup.object({
  email: Yup.string().required("Enter email"),
  password: Yup.string().required(
    "Enter your password. This field can't be empty"
  ),
});

function SignIn() {
  const [showPassword, setShowPassowrd] = useState(false);
  const [error, setError] = useState(null);
  const { logIn } = useContext(Context);

  function showPasswordBtn() {
    setShowPassowrd(!showPassword);

    var x = document.getElementById("showPass");
    if (x.type === "password") {
      x.type = "text";
    } else {
      x.type = "password";
    }
  }

  const onSubmit = async (values, { setSubmitting }) => {
    setError(null);
    let isSuccess = await axios.post(
      "http://localhost:5000/api/goals/login",
      values
    );

    if (isSuccess.data) {
      logIn();
    } else {
      setError("Invalid email or password");
    }
  };
  return (
    <div className={styles.SignIn}>
      <div className={styles.divRow}>
        <div className={styles.formContainer}>
          <header>Log in</header>
          <Formik
            initialValues={{ email: "", password: "" }}
            onSubmit={onSubmit}
            validationSchema={validationSchema}
          >
            <Form className={styles.formInputs}>
              <div className={styles.inputContainer}>
                <Field type="text" name="email" placeholder="Email" />
                <ErrorMessage name="email" />
              </div>
              <div className={styles.inputContainer}>
                <div className={styles.inputPassword}>
                  <Field
                    type="password"
                    name="password"
                    id="showPass"
                    placeholder="Password"
                  />
                  <button type="button" onClick={showPasswordBtn}>
                    {showPassword ? "hide" : "show"}
                  </button>
                </div>
                <ErrorMessage name="password" />
                <div>
                  {error !== null ? (
                    <p className={styles.error}>{error}</p>
                  ) : null}
                  <Link
                    to="/recover-password"
                    className={styles.recoverPassword}
                  >
                    Don't remember the password?
                  </Link>
                </div>
              </div>
              <div className={styles.inputContainer}>
                <button type="submit" className={styles.btnSignIn}>
                  Log in
                </button>
              </div>
            </Form>
          </Formik>
        </div>
        <div className={styles.additionInfo}>
          <RegisterReference />
          <Benefits />
        </div>
      </div>
    </div>
  );
}

export default SignIn;
