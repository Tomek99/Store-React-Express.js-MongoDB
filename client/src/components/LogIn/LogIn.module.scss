.SignIn {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100vh;
    background-color: var(-bg);
    color: #fff;
    margin: 10px;

    .divRow {
        display: flex;
        flex-wrap: wrap;
        max-width: 1620px;
        flex-basis: 100%;
        justify-content: space-evenly;

        .formContainer {
            flex-basis: 500px;
            display: flex;
            flex-direction: column;
            justify-content: center;
            align-items: stretch;
            gap: 20px;
            padding: 20px;
            margin-bottom: 30px;
            background-color: var(--black);
            border: var(--border);
            border-radius: 20px;

            header {
                font-size: 3rem;
                margin-bottom: 20px;
                text-align: center;
            }

            .formInputs {
                display: flex;
                flex-direction: column;
                align-items: stretch;
                gap: 25px;
                margin: 0 30px 0 30px;

                input {
                    width: 100%;
                    border-radius: 16px;
                    padding: 12px 0px 12px 10px;
                    font-size: 1.3rem;
                    border: var(--border);
                    background-color: var(--black);
                    color: #fff;
                }

                .inputPassword {
                    display: flex;

                    input {
                        border-radius: 16px 0 0 16px;
                        border-right: none;
                    }

                    button {
                        padding-right: 10px;
                        border: var(--border);
                        border-left: none;
                        border-radius: 0 16px 16px 0;
                        background: transparent;
                        color: #ccc;
                        cursor: pointer;
                    }
                }

                .btnSignIn {
                    width: 100%;
                    border-radius: 16px;
                    padding: 10px 0px 10px 10px;
                    font-size: 1.8rem;
                    background: var(--main-color);
                    color: #fff;
                    cursor: pointer;
                }
                .error {
                    font-size: 1.2rem;
                    margin-top: 5px;
                    margin-left: 5px;
                    color: rgb(255, 0, 34);
                    font-weight: 300;
                }

                .recoverPassword {
                    text-transform: none;
                    font-size: 1.2rem;
                    margin-left: 5px;
                    color: var(--main-color);

                    &:hover {
                        color: var(--main-color-darken);
                    }
                }
            }
        }
    }

    // .additionInfo {
    //     padding: 20px;
    // }
}

@media (max-width: 1080px) {
    .formContainer {
        header {
            margin-top: 20px;
        }

        .formInputs {
            margin-bottom: 20px !important;
        }
    }
}
@media (max-width: 500px) {
    .SignIn {
        align-items: flex-start;
    }
}
