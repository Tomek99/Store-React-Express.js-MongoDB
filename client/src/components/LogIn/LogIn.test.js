import { render } from "@testing-library/react";
import LogIn from "./LogIn";
import { MemoryRouter } from "react-router-dom";
import { Context } from "../../Contexts/Context";
test("should render Login component", () => {
  const mock = jest.fn();
  const { container } = render(
    <MemoryRouter>
      <Context.Provider value={{}}>
        <LogIn />
      </Context.Provider>
    </MemoryRouter>
  );

  expect(container).toMatchSnapshot();
});

// test("logIn function be should called", () => {
//   const mock = jest.fn()
//   const { container } = render(
//     <MemoryRouter>
//       <Context.Provider value={{}}>
//       <LogIn />
//       </Context.Provider>
//     </MemoryRouter>
//   );

//   expect(container).toMatchSnapshot();
// });
