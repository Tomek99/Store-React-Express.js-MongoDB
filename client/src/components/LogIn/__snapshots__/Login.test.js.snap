// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Login component 1`] = `
<div>
  <div
    class="SignIn"
  >
    <div
      class="divRow"
    >
      <div
        class="formContainer"
      >
        <header>
          Log in
        </header>
        <form
          action="#"
          class="formInputs"
        >
          <div
            class="inputContainer"
          >
            <input
              name="email"
              placeholder="Email"
              type="text"
              value=""
            />
            <div
              class="errorComponent"
            />
          </div>
          <div
            class="inputContainer"
          >
            <div
              class="inputPassword"
            >
              <input
                id="showPass"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
              <button
                type="button"
              >
                show
              </button>
            </div>
            <div
              class="errorComponent"
            />
            <div>
              <a
                class="recoverPassword"
                href="/recover-password"
              >
                Don't remember the password?
              </a>
            </div>
          </div>
          <div
            class="inputContainer"
          >
            <button
              class="btnSignIn"
              type="submit"
            >
              Log in
            </button>
          </div>
        </form>
      </div>
      <div
        class="additionInfo"
      >
        <div
          class="RegisterReference"
        >
          <header>
            Don't have an account yet?
          </header>
          <a
            class="btnReference"
            href="/sign-up"
          >
            Create an account
          </a>
          <button />
        </div>
        <div
          class="Benefits"
        >
          <h1>
            Why you should create an account on cofee shop
          </h1>
          <p>
            <svg
              fill="none"
              height="50"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M5 8h8.5a2.5 2.5 0 1 0 -2.34 -3.24"
              />
              <path
                d="M3 12h15.5a2.5 2.5 0 1 1 -2.34 3.24"
              />
              <path
                d="M4 16h5.5a2.5 2.5 0 1 1 -2.34 3.24"
              />
            </svg>
             
            <span>
              Order faster
            </span>
          </p>
          <p>
            <svg
              fill="currentColor"
              height="50"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.5 3a.5.5 0 0 1 .5.5v9a.5.5 0 0 1-.5.5h-13a.5.5 0 0 1-.5-.5v-9a.5.5 0 0 1 .5-.5h13zm-13-1A1.5 1.5 0 0 0 0 3.5v9A1.5 1.5 0 0 0 1.5 14h13a1.5 1.5 0 0 0 1.5-1.5v-9A1.5 1.5 0 0 0 14.5 2h-13z"
              />
              <path
                d="M7 5.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm-1.496-.854a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 1 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0zM7 9.5a.5.5 0 0 1 .5-.5h5a.5.5 0 0 1 0 1h-5a.5.5 0 0 1-.5-.5zm-1.496-.854a.5.5 0 0 1 0 .708l-1.5 1.5a.5.5 0 0 1-.708 0l-.5-.5a.5.5 0 0 1 .708-.708l.146.147 1.146-1.147a.5.5 0 0 1 .708 0z"
              />
            </svg>
            <span>
              Check previous orders
            </span>
          </p>
          <p>
            <svg
              fill="currentColor"
              height="50"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M1 4.27v7.47c0 .45.3.84.75.97l6.5 1.73c.16.05.34.05.5 0l6.5-1.73c.45-.13.75-.52.75-.97V4.27c0-.45-.3-.84-.75-.97l-6.5-1.74a1.4 1.4 0 0 0-.5 0L1.75 3.3c-.45.13-.75.52-.75.97zm7 9.09l-6-1.59V5l6 1.61v6.75zM2 4l2.5-.67L11 5.06l-2.5.67L2 4zm13 7.77l-6 1.59V6.61l2-.55V8.5l2-.53V5.53L15 5v6.77zm-2-7.24L6.5 2.8l2-.53L15 4l-2 .53z"
                fill-rule="evenodd"
              />
            </svg>
            <span>
              Track your order status
            </span>
          </p>
          <p>
            <svg
              fill="none"
              height="50"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M6 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M17 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0"
              />
              <path
                d="M17 17h-11v-14h-2"
              />
              <path
                d="M20 6l-1 7h-13"
              />
              <path
                d="M10 10l6 -6"
              />
              <path
                d="M10.5 4.5m-.5 0a.5 .5 0 1 0 1 0a.5 .5 0 1 0 -1 0"
              />
              <path
                d="M15.5 9.5m-.5 0a.5 .5 0 1 0 1 0a.5 .5 0 1 0 -1 0"
              />
            </svg>
            <span>
              Benefit from discounts and special bargains
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
